### PR TITLE
Fix validation trigger

### PR DESCRIPTION
**Pull request by jorgepiloto**
Yet another try to fix the validation trigger, hope this to be the last one 🤞🏽

### CODE REVIEW

File: .github/workflows/validation_trigger.yml [hunk 1]
```diff
@@ -26,5 +26,5 @@ jobs:
         run: |
           curl -XPOST -u "${{ secrets.PAT_VALIDATION}}:${{secrets.PAT_VALIDATION_TOKEN}}" \
           -H "Accept: application/vnd.github.v3+json" \
-          -H "Content-Type: application/json" https://api.github.com/repos/poliastro/validation/actions/workflows/ci_actions.yml/dispatches \
+          -H "Content-Type: application/json" https://api.github.com/repos/poliastro/validation/actions/workflows/ci_actions/dispatches \
```
Comment (astrojuanlu) [inline .github/workflows/validation_trigger.yml:29]: I think this is the correct workflow: https://api.github.com/repos/poliastro/validation/actions/workflows/ci_actions.yml

**Review by astrojuanlu — CHANGES_REQUESTED** [review]
It's weird that this still doesn't work 🤔 We would need to find a way so that, if the POST request fails, the workflow fails.

**Comment by jorgepiloto** [comment]
This is working through command line but not sure why it fails if triggered from here... 🤔

File: .github/workflows/validation_trigger.yml [hunk 2]
```diff
@@ -18,13 +18,14 @@ jobs:
   # Validate poliastro's new features
   validation:
     runs-on: ubuntu-latest
-    environment: validation-env
```
Comment (astrojuanlu) [inline .github/workflows/validation_trigger.yml:21]: Shouldn't we keep this? Perhaps if we remove the line, the env vars are not properly read

**Comment by astrojuanlu** [comment]
Idea: use RequestBin or similar services to understand if the `POST` request is being performed correctly. On it.

**Comment by astrojuanlu** [comment]
@jorgepiloto Authorization headers are empty.

File: .github/workflows/validation_trigger.yml [hunk 3]
```diff
@@ -24,7 +24,8 @@ jobs:
     steps:
       - name: Trigger poliastro/validation
         run: |
-          curl -XPOST -u "${{ secrets.PAT_VALIDATION}}:${{secrets.PAT_VALIDATION_TOKEN}}" \
+          curl -v \
+          -XPOST -u "${{ secrets.VALIDATION_USER }}:${{ secrets.VALIDATION_PASS }}" \
```
Comment (astrojuanlu) [inline .github/workflows/validation_trigger.yml:28]: I think this syntax just doesn't work: https://docs.github.com/en/actions/reference/encrypted-secrets#example-using-bash let me try something different

**Comment by astrojuanlu** [comment]
> Secrets are not passed to workflows that are triggered by a pull request from a fork.

Screenshot_2021-04-18 Build software better, together

I think this is it @jorgepiloto. We should find another way to trigger this, I think.

**Review by astrojuanlu — CHANGES_REQUESTED** [review]
After reading https://blog.teddykatz.com/2021/03/17/github-actions-write-access.html, it makes a lot of sense that secrets are not available for pull request, since an attacker could send a malicious pull request and print them.

I am not sure how was this working before, perhaps they changed something. As the action was not failing when an error message is returned, I am not sure when this change happened.

@jorgepiloto although it's very attractive to know whether a pull request passed the validation, I don't think this is possible with our current structure. Given that the token must have `actions:write` permissions on the `repo` token, it's just dangerous to make it public.

I only see these solutions:

- Bring everything into the same repository. I don't think it's very practical, we could discuss it again as a last resort.
- Create our own bot or action that does this properly on pull requests. A lot of work.
- Assume that we will only trigger validations from `main`, and keep an eye on it especially when a release approaches. I think this is the simplest solution for now.

Therefore, we need to make these changes:

- Make the action fail if curl (✔️  done)
- Rework the CI so these triggers are only sent from `main`, where the environment secrets are available, so we avoid getting failures for every pull request
- Squash all the commits into one

**Comment by jorgepiloto** [comment]
Closing this in favor of #1179. For future modifications and maintenance, it is important to remember that this must be done through branches which belong to the repository, not by ones which belong to forks.

**Pull request closed without merging**